Allow ElasticSearch check to override the name it reports itself as

ashb opened the pull request:
This means that instead of seeing:

> ElasticSearchCheck OK: Number of 5XX responses over last hour

we can change it to be:

> App500Errors OK: Number of 5XX responses over last hour

It's a small change but it makes it a lot easier to see what the problem
is, espeically when the alert is going via PagerDuty

(This was discussed in #70)

ashb commented:
Don't merge yet.

SteveMarshall commented on sensu/files/plugins/check-elastic.rb:
```diff
@@ -120,6 +139,10 @@ def run_threshold_check
     count = data['count']
 
+    # Override the check name to be more specific as requested
+    if config[:name] != nil
```
Isn't this the same as `if config[:name]`?

ashb replied:
It behaves differently in the case of an empty string, so probably not all that useful.

SteveMarshall replied:
Does it? The empty string is truthy in Ruby, isn't it?

SteveMarshall commented:
Is it worth describing the other options (e.g. `:result_key`)?

ashb commented:
@SteveMarshall I looked and found that `result_key` is not actually used. So I removed it :D

I also changed the name of the tag param to be less confusing

ashb commented:
I'm happy with this now.
